service/version.go

```diff
@@ -21,14 +21,26 @@ func (uis *UIServer) versionPage(w http.ResponseWriter, r *http.Request) {
 	projCtx := MustHaveProjectContext(r)
 	project, err := projCtx.GetProject()
 
-	if err != nil || project == nil || projCtx.Version == nil {
-		http.Error(w, "not found", http.StatusNotFound)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
I don't think this would resolve the bug because `err` will probably be set if `projCtx` or `project` is `nil`. You should move the redirect above this line to avoid hitting this case entirely. 
https://github.com/evergreen-ci/evergreen/blob/cd6a9f2a659d8822f83a84086318e1d45f49b8cf/model/context.go#L80

This function gets called by GetProject() 
https://github.com/evergreen-ci/evergreen/blob/ecd41df3feffef24ceb380ad5ff5750069660b07/model/project.go#L1269
I didn't look at the actual query but based on the comment I think it is a likely culprit. There would be no mainline version for the project identifier linked to the problematic version. Because it is tracking a branch and not the mainline branch.

Reply:
updated